### ISSUE TITLE
RUF100: have option to require codes, but for codes to be unknown to `ruff`

### ISSUE THREAD

**Issue by eriknw** [charliermarsh/ruff#1882]
This would allow `ruff` to "play nicely" with other linters that projects are still using as they're transitioning to `ruff`.

For example:
```python
    string = "quoteme"
    quoted = f'"{string}"'  # noqa: B028
```
when using https://github.com/PyCQA/flake8-bugbear

I've had this come up with other error codes that `ruff` does not yet know about.

I still want the `RUF100` check (or similar) to _require_ that `noqa` pragmas have a code, but not require that `ruff` know about the codes.

**Comment by charliermarsh** [comment]
I believe we actually support this -- have you checked out the [`external`](https://github.com/charliermarsh/ruff#external) option? It lets you define codes that should effectively be ignored when enforcing RUF100 (for this exact purpose).

**Comment by charliermarsh** [comment]
(But, of course, let me know if I'm misunderstanding the request.)

**Comment by eriknw** [comment]
Perfect, thanks!

I thought I was probably missing something, but I also thought I adequately read/searched the docs.

Closing, and maybe this issue will serve as breadcrumbs for somebody else searching for this :)

**Comment by charliermarsh** [comment]
No prob -- I'll be the first to admit that the docs are unwieldy right now!
